src/virt.c

```diff
@@ -2230,33 +2245,23 @@ static int persistent_domains_state_notification(void) {
 }
 
 static int lv_read(user_data_t *ud) {
-  time_t t;
-  struct lv_read_instance *inst = NULL;
-  struct lv_read_state *state = NULL;
-
   if (ud->data == NULL) {
     ERROR(PLUGIN_NAME " plugin: NULL userdata");
     return -1;
   }
 
-  inst = ud->data;
-  state = &inst->read_state;
-
-  bool reconnect = conn == NULL ? true : false;
-  /* event implementation must be registered before connection is opened */
-  if (inst->id == 0) {
-    if (!persistent_notification && reconnect)
-      if (register_event_impl() != 0)
-        return -1;
+  struct lv_read_instance *inst = ud->data;
+  struct lv_read_state *state = &inst->read_state;
 
+  if (inst->id == 0)
     if (lv_connect() < 0)
       return -1;
 
-    if (!persistent_notification && reconnect && conn != NULL)
-      if (start_event_loop(&notif_thread) != 0)
-        return -1;
-  }
+  /* Wait until inst#0 establish connection */
```

Comment:
does this deserve a [debug] log entry?

Reply:
On my server I have following output:

```
May 13 01:23:02 xxxx systemd[1]: Started Statistics collection and monitoring daemon.
May 13 01:23:02 xxxx collectd[1804]: libvirt: XML-RPC error : Failed to connect socket to '/var/run/libvirt/libvirt-sock-ro': No such file or directory
May 13 01:23:02 xxxx collectd[1804]: virt plugin: Unable to connect: virConnectOpen failed.
May 13 01:23:02 xxxx collectd[1804]: virt plugin: reader virt-0 initialized
May 13 01:23:02 xxxx collectd[1804]: libvirt: XML-RPC error : Failed to connect socket to '/var/run/libvirt/libvirt-sock-ro': No such file or directory
May 13 01:23:02 xxxx collectd[1804]: Initialization complete, entering read-loop.
May 13 01:23:02 xxxx collectd[1804]: read-function of plugin `virt-0' failed. Will suspend it for 20.000 seconds.
May 13 01:23:23 xxxx collectd[1804]: virt plugin: Connection established.
```

Of course, `virt-X` read function calls will be hidden. 
I have no strong opinion if we need proposed [debug] entry. I can add it if  you think it is needed.

Reply:
`DEBUG(PLUGIN_NAME " plugin#%s: Wait until inst#0 establish connection", inst->tag);` will be good? I add this to push CI ;-)

Reply:
Yep, good for me